src/animation/AnimationClip.js

```diff
@@ -244,60 +245,47 @@ THREE.AnimationClip.parseAnimation = function( animation, bones, nodeName ) {
 			}
 
 			// create a track for each morph target with all zero morphTargetInfluences except for the keys in which the morphTarget is named.
-			for ( var morphTargetName in morphTargetNames ) {
+			for( var morphTargetName in morphTargetNames ) {
 
-				var keys = [];
+				var times = [];
+				var values = [];
 
-				for ( var m = 0; m < animationKeys[k].morphTargets.length; m ++ ) {
+				for( var m = 0; m < animationKeys[k].morphTargets.length; m ++ ) {
 
 					var animationKey = animationKeys[k];
 
-					keys.push( {
-							time: animationKey.time,
-							value: (( animationKey.morphTarget === morphTargetName ) ? 1 : 0 )
-						});
+					times.push( animationKey.time );
+					values.push( ( animationKey.morphTarget === morphTargetName ) ? 1 : 0 )
 
 				}
 
-				tracks.push( new THREE.NumberKeyframeTrack( nodeName + '.morphTargetInfluence[' + morphTargetName + ']', keys ) );
+				tracks.push( new THREE.NumberKeyframeTrack( nodeName + '.morphTargetInfluence[' + morphTargetName + ']', times, values ) );
 
 			}
 
 			duration = morphTargetNames.length * ( fps || 1.0 );
 
-		} else {
+		}
+		else {
 
 			var boneName = nodeName + '.bones[' + bones[ h ].name + ']';
 
 			// track contains positions...
-			var positionTrack = convertTrack( boneName + '.position', animationKeys, 'pos', THREE.VectorKeyframeTrack, function( animationKey ) {
-					return new THREE.Vector3().fromArray( animationKey.pos )
-				} );
-
-			if ( positionTrack ) tracks.push( positionTrack );
+			var positionTrack = convertTrack( boneName + '.position', animationKeys, 'pos', THREE.VectorKeyframeTrack );
+			if( positionTrack ) tracks.push( positionTrack );
 
 			// track contains quaternions...
-			var quaternionTrack = convertTrack( boneName + '.quaternion', animationKeys, 'rot', THREE.QuaternionKeyframeTrack, function( animationKey ) {
-					if ( animationKey.rot.slerp ) {
-						return animationKey.rot.clone();
-					} else {
-						return new THREE.Quaternion().fromArray( animationKey.rot );
-					}
-				} );
-
-			if ( quaternionTrack ) tracks.push( quaternionTrack );
+			var quaternionTrack = convertTrack( boneName + '.quaternion', animationKeys, 'rot', THREE.QuaternionKeyframeTrack );
+			if( quaternionTrack ) tracks.push( quaternionTrack );
 
 			// track contains quaternions...
-			var scaleTrack = convertTrack( boneName + '.scale', animationKeys, 'scl', THREE.VectorKeyframeTrack, function( animationKey ) {
-					return new THREE.Vector3().fromArray( animationKey.scl )
-				} );
-
-			if ( scaleTrack ) tracks.push( scaleTrack );
+			var scaleTrack = convertTrack( boneName + '.scale', animationKeys, 'scl', THREE.VectorKeyframeTrack );
+			if( scaleTrack ) tracks.push( scaleTrack );
 
 		}
 	}
 
-	if ( tracks.length === 0 ) {
+	if( tracks.length === 0 ) {
```

Comment:
This is actually code I didn't touch. The diff is against a tidier version of the code that I had to revert when pulling up...